sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/GpuIntervalUtils.scala

```diff
@@ -292,17 +292,16 @@ object GpuIntervalUtils extends Arm {
   // not close firstSignInTable and secondSignInTable here, outer table.close will close them
   private def finalSign(
       firstSignInTable: ColumnVector, secondSignInTable: ColumnVector): ColumnVector = {
-    withResource(Scalar.fromString("-")) { negScalar =>
-      withResource(negScalar.equalTo(firstSignInTable)) { neg1 =>
-        withResource(negScalar.equalTo(secondSignInTable)) { neg2 =>
-          withResource(neg1.bitXor(neg2)) { s =>
-            withResource(Scalar.fromLong(1L)) { one =>
-              withResource(Scalar.fromLong(-1L)) { negOne =>
-                s.ifElse(negOne, one)
-              }
-            }
-          }
-        }
+    val negatives = withResource(Scalar.fromString("-")) { negScalar =>
+      lazyReduce(Seq(
+        lazyResource(negScalar.equalTo(firstSignInTable)),
+        lazyResource(negScalar.equalTo(secondSignInTable))
+      ))(_ bitXor _)
```

Comment:
> isn't negScalar closed before it's accessed here because.

yes, this is a bug in here. Mixing with the old idiom is a challenge. 

> I'm not a fan of the lazyResource / lazyReduce idiom.

Tabling the API until / unless it becomes more convincing.